moduleframework/tools/dockerlint.py

```diff
@@ -97,6 +99,81 @@ def test_helpmd_is_present(self):
         self.assertTrue(self.dp.check_helpmd_is_present())
 
 
+class DockerfileLinterInContainer(container_avocado_test.ContainerAvocadoTest):
+    """
+    :avocado: enable
+
+    """
+
+    def _file_to_check(self, doc_file_list):
+        test_failed = False
+        for doc in doc_file_list.split('\n'):
+            exit_status = self.run("test -e %s" % doc, ignore_status=True).exit_status
+            if int(exit_status) == 0:
+                self.log.debug("%s doc file exists in container" % doc)
+                test_failed = True
+        return test_failed
+
+    def test_all_nodocs(self):
+        """
+        This test checks all docs installed in container. As delivered by Platform
+        as delivered by installing packages RUN command
+        :return: WARNING in case we found some docs
+                 PASS in case there is no doc file found
+        """
+        self.start()
+        all_docs = self.run("rpm -qad", verbose=False).stdout
+        test_failed = self._file_to_check(all_docs)
+        if test_failed:
+            self.log.warn("Documentation files exist in container. They are installed by Platform or by RUN commands.")
+        self.assertTrue(True)
+
+    def test_installed_docs(self):
+        """
+        This test checks whether no docs are installed by RUN dnf command
+        :return: FAILED in case we found some docs
+                 PASS in case there is no doc file found
+        """
+        self.start()
+        # Double brackets has to by used because of trans_dict.
+        # 'EXCEPTION MTF: ', 'Command is formatted by using trans_dict.
+        # If you want to use brackets { } in your code, please use {{ }}.
+        installed_pkgs = self.run("rpm -qa --qf '%{{NAME}}\n'", verbose=False).stdout
+        defined_pkgs = self.backend.getPackageList()
+        list_pkg = set(installed_pkgs).intersection(set(defined_pkgs))
+        test_failed = False
+        for pkg in list_pkg:
+            pkg_doc = self.run("rpm -qd %s" % pkg, verbose=False).stdout
+            if self._file_to_check(pkg_doc):
+                test_failed = True
+        self.assertFalse(test_failed)
+
+    def test_docker_clean_all(self):
+        """
+        This test checks if size of /var/cache/<pkg_manager> is bigger then
+        150000 taken by command du -hsb /var/cache/<pkg_manager>
```

Comment:
Why 150000? I just tried creating an image where I installed a package from random copr repo and even though I did `dnf clean all`, the directory had `6537257`, so it clearly wouldn't pass this check even though it's correct.

Reply:
This is a bit tricky. We have no possibility, how to test if cache is really clean. We have to figure out Fedora and RHEL systems.

Reply:
I just made that number up in one of previous comments - when I populated cache from bigger repos, the cache had tenths of megabytes and when I did `clean all` there were still some files left. So maybe 10kB could be the threshold?

Reply:
I haven't looked at yum, but for dnf cache, you could check that none of these files exist:

* `/var/cache/dnf/*.solv`
* `/var/cache/dnf/*.solvx`
* `/var/cache/dnf/*/metalink.xml`
* `/var/cache/dnf/*/repodata/*.xml.gz`
* `/var/cache/dnf/*/packages/*.rpm`

These are the files that seem to be deleted when you run `dnf clean all`. I'm sure you can find similar behaviour for yum. I think it'd be much better checking that none of these files exist instead of comparing directory size to a magic number that doesn't even work as I've demonstrated above.

Reply:
Good point, we can separate it. If yum is used, then set threshold to 10000, in case dnf check *.solv* files. It makes more sentence then the current implementation.

Reply:
Why would you want to use threshold for yum? Isn't it possible to use something similar to the `.solv` files for yum?

Reply:
@bkabrda I tried it back then and it wasn't easy to figure out